.github/workflows/hil.yml

```diff
@@ -95,24 +98,45 @@ jobs:
       matrix:
         target:
           # RISC-V devices:
-          # - soc: esp32c3
-          #   runner: rustboard
+          - soc: esp32c3
+            runner: esp32c3-usb
           - soc: esp32c6
             runner: esp32c6-usb
-          # - soc: esp32h2
-          #   runner: esp32h2-usb
-          # # Xtensa devices:
-          # - soc: esp32s3
-          #   runner: esp32s3-usb
+          - soc: esp32h2
+            runner: esp32h2-usb
+          # Xtensa devices:
+          - soc: esp32s2
+            runner: esp32s2-jtag
+          - soc: esp32s3
+            runner: esp32s3-usb
     steps:
       - uses: actions/download-artifact@v4
         with:
           name: tests-${{ matrix.target.soc }}
-          path: tests
-      - name: Run tests
+          path: tests-${{ matrix.target.soc }}
+
+      - name: Run Tests
         run: |
```

Comment:
Would really like to see this implemented in the `xtask` instead, please.

Reply:
Me too, this bash code and the one preparing the artifacts looks quite bad to me. Any suggestions on how to integrate this with `xtask` other than creating a subcommand that points to a folder, and it executes all the elfs on that folder?

Reply:
Added a subcommand to run all the elfs in a folder in [`daf8746`](https://github.com/esp-rs/esp-hal/pull/1513/commits/daf874691664ef5ecfdb3f2c9f09579120213534). Here is a HIL run: https://github.com/esp-rs/esp-hal/actions/runs/8924001488 with that changes, it increases the time it takes for the run as it needs to build the xtask package.